### ISSUE TITLE
Allow for filter expressions to exclude unmapped fields instead of throwing an exception

### ISSUE THREAD

**Issue by riccardorestagno** [alirezanet/Gridify#34]
When upgrading from version 1 to 2, I noticed that if the filter expression contains a column name that cannot be found in the database mapping, the method throws a GridifyFilteringException ("Invalid expression"). In version 1 this was not the case, and the incorrect filter value would simply be skipped when creating the query.

Would it be possible to skip adding the filtering to the column if it was not mapped?

I have fixed the underlying issue on my side by ensuring only mapped column names are included in the filter expression, however I think allowing for graceful fails by simply skipping the unmapped filter expression is safer than throwing an exception.

**Comment by alirezanet** [comment]
Hello,
I think depending on the use case both behaviors make sense, I'll probably add a configuration to give you control to choose.

**Comment by alirezanet** [comment]
In case we have unmapped filtering what do you expect, returning all the records or returning no records?
```c#
      [Fact] // issue #34
      public void ApplyFiltering_UnmappedFields_ShouldSkipWhenIgnored()
      {
         var gm = new GridifyMapper<TestClass>(configuration => configuration.IgnoreNotMappedFields = true)
            .AddMap("Id", q => q.Id);
      
         // name=*a filter should be ignored
         var actual = _fakeRepository.AsQueryable().ApplyFiltering("name=*a, id>15", gm).ToList();
         var expected = _fakeRepository.Where(q => q.Id > 15).ToList();
      
         Assert.Equal(expected.Count, actual.Count);
         Assert.Equal(expected, actual);
         Assert.True(actual.Any());
      }
```

**Comment by alirezanet** [comment]
Is this feature what you need? 
e.g
```c#
      [Fact] // issue #34
      public void ApplyOrdering_UnmappedFields_ShouldSkipWhenIgnored()
      {
         var gm = new GridifyMapper<TestClass>(configuration => configuration.IgnoreNotMappedFields = true)
            .AddMap("Id", q => q.Id);
      
         // name=*a filter should be ignored
         var actual = _fakeRepository.AsQueryable().ApplyOrdering("name,id", gm).ToList();
         var expected = _fakeRepository.OrderBy(q => q.Id ).ToList();
      
         Assert.Equal(expected.Count, actual.Count);
         Assert.Equal(expected, actual);
         Assert.True(actual.Any());
      }
```
I've added a new configuration to the GridifyMapper that you can use to prevent throwing exceptions if you had unmapped fields. `IgnoreNotMappedFields`

**Comment by riccardorestagno** [comment]
Yes that would be great! Thanks!

**Comment by alirezanet** [comment]
this feature is available in version `v2.3.2`.
also, I change the error in case a field is not mapped by default. instead of `GridifyFilteringException` it throws `GridifyMapperException`.
